### PR TITLE
Radio test regulatory support and antenna gain configuration support

### PR DESCRIPTION
**Pull request by sachinthegreen**
- Add regulatory support to the Radio Test sample
- Add antenna gain configuration support

### CODE REVIEW

**Review by srkanordic — APPROVED** [review]

File: drivers/wifi/nrf700x/Kconfig.nrf700x [hunk 1]
```diff
@@ -27,6 +27,11 @@ if WIFI_NRF700X
 config NRF700X_TCP_IP_CHECKSUM_OFFLOAD
 	bool "Enable TCP/IP checksum offload"
 
+config NRF700X_REG_DOMAIN
+	string "The ISO/IEC alpha2 country code for the country in which this device is currently operating. Default 00 (World regulatory)"
+# 00 is used for World regulatory
```
Comment (krish2718) [inline drivers/wifi/nrf700x/Kconfig.nrf700x:32]: As this is covered in help, comment is redundant.

**Review by krish2718 — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]

File: samples/wifi/radio_test/sample_description.rst [hunk 2]
```diff
@@ -558,8 +559,56 @@ Testing
...
+               reg_domain = US
+               bypass_reg_domain = 1
+
+         .. note::
+
+            Bypass regulatory domain is false by default.
+
+            If ``bypass_reg_domain`` is ``0``, then TX power of the channel will be configured to
+            the minimum value of the user configured TX power value and maximum power supported in the
+            configured regulatory domain.
```
Comment (richabp) [inline samples/wifi/radio_test/sample_description.rst:604]: It seems there is some unnecessary space within the parts of the sentence.
```suggestion
            If ``bypass_reg_domain`` is ``0``, then TX power of the channel will be configured to the minimum value of the user configured TX power value and maximum power supported in the configured regulatory domain.
```

File: samples/wifi/radio_test/sample_description.rst [hunk 3]
```diff
@@ -558,8 +559,56 @@ Testing
...
+
+            If ``bypass_reg_domain`` is ``1``, then user configured TX power value will be set
+            overriding current configured regulatory domain max TX power for the channel.
```
Comment (richabp) [inline samples/wifi/radio_test/sample_description.rst:607]:
```suggestion
            If ``bypass_reg_domain`` is ``1``, then user configured TX power value will be set overriding current configured regulatory domain max TX power for the channel.
```

Comment (richabp) [inline samples/wifi/radio_test/sample_description.rst:607]:
```suggestion
            If ``bypass_reg_domain`` is ``1``, then user configured TX power value will be set
            overriding current configured regulatory domain maximum TX power for the channel.
```

**Review by richabp — APPROVED** [review]

**Pull request merged**